### DISCUSSION TITLE
insertContentAt - editor text gets stuck at a certain position

### DISCUSSION THREAD

**Discussion by fbolcic** [ueberdosis/tiptap#4101] [General Questions]
Hello,

We are building a custom node view for our link elements. We got most of the UI nailed down and would like to add a keyboard shortcut for triggering the link and its accompanying edit link modal. You can see a small demo here:

https://github.com/ueberdosis/tiptap/assets/101173923/27068009-8f7c-48b9-8f1b-3ee39f9b0a2c

This is the code we are using for the shortcut:
```
  addKeyboardShortcuts() {
    return {
      'Mod-Shift-u': () => {
        const { from, to } = this.editor.state.selection;

        return this.editor
              .chain()
              .insertContentAt({ from, to }, `<a href="">${this.editor.state.doc.textBetween(from, to)}</a> `)
              .setNodeSelection(from)
              .run();
      },
    };
  },
```

There are a couple of things happening here:
1. We are using `insertContentAt` to insert the node. We would like to use the `setNode` but unfortunately it does not work with inline nodes (such as ours) so we had to improvise.
2. We are taking the selected text content (with the `textBetween` function and current position) and placing it in the `<a></a>` element. After that our node is rendered as a link with the selected text
3. Finally we select the node. By selecting it we control the small edit link modal that is opened in the video.

As you can see from the video, it basically works fine but unfortunately at a certain position in the link (at the end of it to be exact) some text gets stuck and is not really treated in the editor intuitively.
Whatever we write is not appended to the end of the link but gets prepended. Also, the text is not registered when we hook into the `onChange` and can't be seen as a part of the editor JSON content. It's like an editable ghost inside the editor 👻 .

What are we doing wrong here? 
Do you have some ideas for a better approach in our problem?

**Comment by fbolcic** [comment]
Hello again.
I figured out that the problem is not in the `insertContentAt` function per se, but the edited text somehow gets added to the node view wrapper - and it seems like the editor does not register it anymore. This can be seen in the following vide:

https://github.com/ueberdosis/tiptap/assets/101173923/da32cc39-522d-4858-885b-3775ae6f44e2

If anyone has any ides how to solve this please help!

**Reply by emrullahtastan** [reply]
@fbolcic hello my friend. 

Also I want to do that. So, I want to give a link to selection text with tooltip or bubble menu. How I do this? Actually, I can make when learn a selection text position on DOM. But I don't learn that.

Thank you very much for your supporting.

**Reply by emrullahtastan** [reply]
I resolved. Thank you.

**Comment by fbolcic** [comment]
Bumping this discussion since we are still stuck with this feature.
Can we get any help from the maintainers please? A simple nudge in the right direction would be greatly appreciated.

**Reply by semoal** [reply]
Did you figure it out? We're stuck with the same problem; we want to render a react custom view, but the result will be just text, once it's rendered for the first time, it is not selectable, and the bubbble menu doesn't appear like it doesn't exist

**Reply by fbolcic** [reply]
No, since we had no response from the maintainers we had to postpone the feature. 
Let us know if you manage to find a solution.
